settings/memcache.settings.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * @file
+ * Contains configuration for the Drupal 8 Memcache API and Integration and
+ * settings for using Memcached on Acquia Cloud.
+ *
+ * Note: the memcache module must be enabled before using memcache as a backend.
+ *
+ * @see: https://api.drupal.org/api/drupal/core%21core.api.php/group/cache/8.4.x
+ * @see: https://www.drupal.org/node/2754947
+ * @see: http://cgit.drupalcode.org/memcache/tree/README.txt?h=8.x-2.x
+ * @see: https://docs.acquia.com/cloud/performance/memcached
+ */
+
+use Drupal\Core\Config\BootstrapConfigStorageFactory;
+use Drupal\Core\Database\Database;
+
+if ($is_ah_env) {
+  // Note that this connects the database, so has to go after any
+  // $databases definitions. See: https://www.drupal.org/node/2766509
+  Database::setMultipleConnectionInfo($databases);
+  $bootstrap_config = BootstrapConfigStorageFactory::get($class_loader);
+  $modules = $bootstrap_config->read('core.extension');
+
+  if ($modules && isset($modules['module']['memcache'])) {
+    // Use Memcached extension.
+    $memcached_exists = class_exists('Memcached', FALSE);
+
+    if ($memcached_exists) {
+      $settings['memcache']['extension'] = 'Memcached';
+    }
+
+    // Use memcache as the default bin.
+    $settings['cache']['default'] = 'cache.backend.memcache';
```

Comment:
Should this be contained within the $memcached_exists condition?

Reply:
Not really, this tells Drupal the use the cache backend provided by the Memacahe module. The `memcached_exists` check serves a different purpose -- it selects the PECL library to talk to the `memcached` daemon. PHP 5.6 on ACE is build with both the Memcache and Memcached libraries, while PHP 7.1 only includes Memcached, which is better maintained.